projects/storefrontlib/cms-components/content/video/video.component.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AfterViewChecked,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import {
+  CmsBannerComponentMedia,
+  CmsResponsiveBannerComponentMedia,
+  CmsService,
+  CmsVideoComponent,
+  ContainerSizeOptions,
+  PageType,
+  SemanticPathService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { take, tap } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+import {
+  Media,
+  MediaContainer,
+} from '../../../shared/components/media/media.model';
+import { MediaService } from '../../../shared/components/media/media.service';
+
+@Component({
+  selector: 'cx-video',
+  templateUrl: './video.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VideoComponent implements AfterViewChecked, OnDestroy {
+  @HostBinding('class') styleClasses: string | undefined;
+  @ViewChild('videoPlayer') videoPlayer: ElementRef;
+
+  source: string | undefined;
+  thumbnail: Media | undefined;
+  routerLink: string | any[] | undefined;
+  autoPlay: boolean;
+  loop: boolean;
+  mute: string | undefined;
+  height: string;
+  protected subscriptions = new Subscription();
+
+  data$: Observable<CmsVideoComponent> = this.component.data$.pipe(
+    tap((data) => {
+      this.styleClasses = data.styleClasses;
+      this.setMedia(data.video, data.videoMedia);
+      this.setControls(data.autoPlay, data.loop, data.mute);
+      this.setVideoHeight(data.containerSize, data.videoContainerHeight);
+      this.setRouting(data);
+    })
+  );
+
+  constructor(
+    protected component: CmsComponentData<CmsVideoComponent>,
+    protected mediaService: MediaService,
+    protected urlService: SemanticPathService,
+    protected cmsService: CmsService,
+    protected cd: ChangeDetectorRef,
+    protected renderer: Renderer2
+  ) {}
+
+  ngAfterViewChecked(): void {
+    this.renderer.setStyle(
```

Comment:
why we have this? we can just add `[height]` in `<video..>`

Reply:
It's because [height] only supports the size in pixels. So we wouldn't be able to set 100% height using that attribute for example.

Maybe there's a better way, let me know if you think there is

Reply:
some thing like this:
```
 [style.height.%]="height === '100%' ? '100' : null"
 [style.height.px]="height !== '100%' ? height : null"
```
if we use this way, remove `px` from `${videoContainerHeight}px`. Or `height` can be either `FULL` or some value `200`, then we check `height === 'FULL' ? 100: null`.

Reply:
Modified approach, let me know what you think.

I call setHeight() directly from the template. If there is no pixel height defined then default will be 100%